### DISCUSSION TITLE
Exclude selected markdown files from blog

### DISCUSSION THREAD

**Discussion by drdv** [squidfunk/mkdocs-material#7780] [General]
Is it possible for used-defined markdown files that are in `blog/posts` to not be parsed for inclusion in the blog? For example, I could have `blog/posts/some-post/README.md` that is meant only for internal documentation purposes.

**Comment by squidfunk** [comment]
It's currently not supported, and I don't think it's easy to pull off with the hacks in the blog plugin that we need to do to make it work with MkDocs. Note that [we're working in the background into making all of this much, much simpler](https://squidfunk.github.io/mkdocs-material/blog/2024/08/19/how-were-transforming-material-for-mkdocs/), including filtering Markdown files in blog posts, so it will be possible in the future. With the current architecture, it might be achievable by writing a custom hook that removes the file from the `Files` collection in `on_files`, but that is only a guess, so some prototyping is necessary. 

I'm short on time right now, but maybe you can share your solution if you find one.

**Reply by drdv** [reply]
My current solution is to use the following tree:
```
[.../blog]>tree
.
├── post1-artifacts-generation-dir
├── post2-artifacts-generation-dir
└── posts
    ├── post1.md
    └── post2.md
```
i.e., simply move the `post*-artifacts-generation-dir` directories out of `blog/posts`. Thinking a bit about all this, probably excluding individual `.md` files is an overkill in my case (and it is actually inconvenient). All that's needed is to have the possibility to somehow signal to `mkdocs` that a given directory (and it content) is not to be processed. One approach that comes to mind is to have a "special file", e.g., `.mkdocsignore` which if present in a directory it is left alone. Frankly, I don't mind putting the artifact-generation dirs one level up (as long as they are in the `blog`) and letting the `posts` keep only interface `.md` files.

Thanks

**Reply by kamilkrzyskow** [reply]
> All that's needed is to have the possibility to somehow signal to mkdocs that a given directory (and it content) is not to be processed

I'm assuming you're looking for the [`draft_docs`](https://www.mkdocs.org/user-guide/configuration/#draft_docs) setting or `exclude_docs`, which take a .gitignore like format to target excluded paths.

Or perhaps you want a the pages to be available remotely in the built site, but "hidden". Then there is [`not_in_nav`](https://www.mkdocs.org/user-guide/configuration/#not_in_nav) + a custom [hook](https://www.mkdocs.org/user-guide/configuration/#hooks), which would adjust the data to remove the pages from the `sitemap.xml` and create a `robots.txt` to exclude the sites from crawlers.

Something like that was implemented in the [`mkdocs-nype`](https://github.com/nypesap/mkdocs-nype) theme as part of the [`nype_tweaks`](https://github.com/nypesap/mkdocs-nype/commit/29c10d4b495c935854d3c5297843462a95030d25#diff-1872fec3e2960be24008470b739e4c34256e43d02d6ccf712e4a16b55eae6fbcR138) plugin, you can extract the logic into a hook.

**Reply by squidfunk** [reply]
@kamilkrzyskow the big problem is that the blog plugin must jump through some hoops to steal the posts from MkDocs, so they are not rendered in the navigation, since blog posts should only be linked from the blog index or archive and category pages. I haven't checked whether this got any easier with MkDocs 1.6.0, since it seems to be easier to at least generate files, but I doubt it. We're working on making this much easier, but with MkDocs current architecture, there's no easy, nor canonical way to do it.

https://github.com/squidfunk/mkdocs-material/blob/b368cee13f23b52335fd1ab39cacae9e26b08291/src/plugins/blog/plugin.py#L171-L208

**Reply by kamilkrzyskow** [reply]
Yeah, I didn't really look into the OP issue myself, as MkDocs expects the navigation to be connected to the filesystem, so there are issues all around because of it. I mostly responded to the comment with the different approach, which puts the non-blog posts in a different directory, so they're not handled the by plugin at all. Everything that's left to do to hide them from the build is to use the `draft_docs`.
